src/test/regress/expected/seclabel.out

```diff
@@ -0,0 +1,165 @@
+---------------------------------------------------------------------
+
+(1 row)
+
+-- with that same connection, we register the label provider in the worker node
+SELECT run_commands_on_session_level_connection_to_node('SELECT citus_test_register_label_provider()');
+ run_commands_on_session_level_connection_to_node
+---------------------------------------------------------------------
+
+(1 row)
+
+SET citus.log_remote_commands TO on;
+SET citus.grep_remote_commands = '%SECURITY LABEL%';
+-- then we run a security label statement which will use the same connection to the worker node
+-- it should finish successfully
+SECURITY LABEL for citus_tests_label_provider ON ROLE user1 IS 'citus_classified';
+NOTICE:  issuing SECURITY LABEL FOR citus_tests_label_provider ON ROLE user1 IS 'citus_classified'
+DETAIL:  on server postgres@localhost:xxxxx connectionId: xxxxxxx
+SECURITY LABEL for citus_tests_label_provider ON ROLE user1 IS NULL;
+NOTICE:  issuing SECURITY LABEL FOR citus_tests_label_provider ON ROLE user1 IS NULL
+DETAIL:  on server postgres@localhost:xxxxx connectionId: xxxxxxx
+SECURITY LABEL for citus_tests_label_provider ON ROLE user1 IS 'citus_unclassified';
+NOTICE:  issuing SECURITY LABEL FOR citus_tests_label_provider ON ROLE user1 IS 'citus_unclassified'
+DETAIL:  on server postgres@localhost:xxxxx connectionId: xxxxxxx
+RESET citus.log_remote_commands;
+SELECT stop_session_level_connection_to_node();
+ stop_session_level_connection_to_node
+---------------------------------------------------------------------
+
+(1 row)
+
+\c - - - :worker_1_port
+SELECT objtype, objname, provider, label FROM pg_seclabels;
+ objtype | objname |          provider          |       label
+---------------------------------------------------------------------
+ role    | user1   | citus_tests_label_provider | citus_unclassified
+(1 row)
+
+\c - - - :master_port
+-- adding a new node will fail because the label provider is not there
+-- however, this is enough for testing as we can see that the SECURITY LABEL commands
+-- will be propagated when adding a new node
+SET citus.log_remote_commands TO on;
+SET citus.grep_remote_commands = '%SECURITY LABEL%';
+SELECT 1 FROM citus_add_node('localhost', :worker_2_port);
+NOTICE:  issuing SELECT worker_create_or_alter_role('user1', 'CREATE ROLE user1 NOSUPERUSER NOCREATEDB NOCREATEROLE INHERIT NOLOGIN NOREPLICATION NOBYPASSRLS CONNECTION LIMIT -1 PASSWORD NULL VALID UNTIL ''infinity''', 'ALTER ROLE user1 NOSUPERUSER NOCREATEDB NOCREATEROLE INHERIT NOLOGIN NOREPLICATION NOBYPASSRLS CONNECTION LIMIT -1 PASSWORD NULL VALID UNTIL ''infinity''');SECURITY LABEL FOR citus_tests_label_provider ON ROLE user1 IS 'citus_unclassified'
+DETAIL:  on server postgres@localhost:xxxxx connectionId: xxxxxxx
+WARNING:  security label provider "citus_tests_label_provider" is not loaded
+CONTEXT:  while executing command on localhost:xxxxx
+ERROR:  failure on connection marked as essential: localhost:xxxxx
```

Comment:
Don't we expect security label to be propagated as a dependency?

Reply:
No, it is not Citus's responsibility to load the security label provider. Normally, the newly added node should have the security label provider already loaded, probably through a module like sepgsql

Reply:
I understood your point now. So the thing is, register_seclabel things must have been called in workers already and EnsureDependencies cannot do much about it anyway, I see.

And we have this error here because we load the seclabel via session level connection.